snooty/postprocess.py

```diff
@@ -976,7 +983,59 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
             assert isinstance(argument, n.Reference)
             source = argument.refuri
 
-        self.openapi_pages[current_slug] = self.SourceData(source_type, source)
+        api_version = node.options.get("api_version", None)
+        resource_versions: Optional[List[str]] = None
+
+        # Fetch OpenAPI versioning data if options are present
+        if api_version and source == "cloud":
+            # Fetch latest git_hash for S3 versioning data
```

Comment:
I have opted for the more efficient fetching option from the Tech Spec. It seems about twice as fast. Not a huge gain, but every bit counts these days. This might mean we do not need to support the :versions: option at all.